tempodb/encoding/common/interfaces.go

```diff
@@ -0,0 +1,53 @@
+package common
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/tempodb/backend"
+)
+
+type Finder interface {
+	FindTraceByID(ctx context.Context, id ID) (*tempopb.Trace, error)
+}
+
+type Searcher interface {
+	Search(ctx context.Context, req *tempopb.SearchRequest, opts SearchOptions) (*tempopb.SearchResponse, error)
+}
+
+type SearchOptions struct {
+	ChunkSizeBytes     uint32 // Buffer size to read from backend storage.
+	StartPage          int    // Controls searching only a subset of the block. Which page to begin searching at.
+	TotalPages         int    // Controls searching only a subset of the block. How many pages to search.
+	MaxBytes           int    // Max allowable trace size in bytes. Traces exceeding this are not searched.
+	PrefetchTraceCount int    // How many traces to prefetch async.
+}
+
+func DefaultSearchOptions() SearchOptions {
+	return SearchOptions{
+		ChunkSizeBytes: 1_000_000,
```

Comment:
i think we're just going to have to have a period in which there are settings and options that don't make sense for parquet.

Reply:
Agree, and I think that's ok.  I'm sure there will be new parquet settings which don't make sense for the existing blocks also.